format.go

```diff
@@ -342,23 +351,46 @@ func (a *unknownAny) ProtoMessage() {
 
 var _ proto.Message = (*unknownAny)(nil)
 
-// RequestParserAndFormatterFor returns a request parser and formatter for the
+// FormatterOptions is a set of flags that are passed to a JSON or text formatter.
+type FormatterOptions struct {
+	EmitJSONDefaultFields bool // FormatJSON only
+	IncludeTextSeparator  bool // FormatText only
+	AllowUnknownFields    bool // FormatJSON only
```

Comment:
Let's group the JSON related fields together, instead of interleaving.

Reply:
Technically `AllowUnknownFields` is an option for the parser, not the formatter.

How about rename this to `FormatOptions` -- since they can apply to parsing the format, not just outputting it.